docs(upgrading/remix): update with latest info (#12310)
remix-run/react-router: docs/upgrading/remix.md

**👉 Update your dependencies (manual)**

If you prefer not to use the codemod, you can manually update your dependencies.

<details>
<summary>Expand to see a table of package name changes in alphabetical order</summary>

| Remix v2 Package                   |     | React Router v7 Package                      |
| ---------------------------------- | --- | -------------------------------------------- |
| `@remix-run/architect`             | ➡️  | `@react-router/architect`                    |
| `@remix-run/cloudflare`            | ➡️  | `@react-router/cloudflare`                   |
| `@remix-run/dev`                   | ➡️  | `@react-router/dev`                          |
| `@remix-run/express`               | ➡️  | `@react-router/express`                      |
| `@remix-run/fs-routes`             | ➡️  | `@react-router/fs-routes`                    |
| `@remix-run/node`                  | ➡️  | `@react-router/node`                         |
| `@remix-run/react`                 | ➡️  | `react-router`                               |
| `@remix-run/route-config`          | ➡️  | `@react-router/dev`                          |

> | `@remix-run/routes-option-adapter` | ➡️  | ` @react-router/remix-routes-option-adapter` |

Reviewer: ```suggestion
| `@remix-run/routes-option-adapter` | ➡️  | `@react-router/remix-routes-option-adapter`  |
```

| `start`     | `remix-serve build/server/index.js` | ➡️  | `react-router-serve build/server/index.js` |
| `typecheck` | `tsc`                               | ➡️  | `react-router typegen && tsc`              |

## 4. Add a `routes.ts` file

<docs-info>

If you used the codemod _and_ Remix v2 `unstable_routeConfig` flag, you can skip this step as it was automatically completed.

</docs-info>

In React Router v7 you define your routes using the `app/routes.ts` file. View the [routing documentation][routing] for more information.

**👉 Update dependencies (if using Remix v2 `unstable_routeConfig` flag)**

```diff
// app/routes.ts
-import { type RouteConfig } from "@remix-run/route-config";
-import { flatRoutes } from "@remix-run/fs-routes";
-import { remixRoutesOptionAdapter } from "@remix-run/routes-option-adapter";
+import { type RouteConfig } from "@react-router/dev/routes";
+import { flatRoutes } from "@react-router/fs-routes";
+import { remixRoutesOptionAdapter } from "@react-router/remix-routes-option-adapter";

> export default {
>   // however your routes are defined
> } satisfies RouteConfig;

Reviewer: ```suggestion
export default [
  // however your routes are defined
] satisfies RouteConfig;
```

```diff
export default defineConfig({
  plugins: [
    remix({
      ssr: true,
-     ignoredRouteFiles: ['**/*'],
-     routes(defineRoutes) {
-       return defineRoutes((route) => {
-         route("/somewhere/cool/*", "catchall.tsx");
-       });
-     },
    })
    tsconfigPaths(),
  ],
});
```

## 5. Add a React Router config

**👉 Add `react-router.config.ts` your project**

The config that was previously passed to the `remix` plugin in `vite.config.ts` is now exported from `react-router.config.ts`.

```shellscript nonumber
touch react-router.config.ts
```

```diff
// vite.config.ts
export default defineConfig({
  plugins: [
-   remix({
-     ssr: true,
-   }),
+   remix(),
    tsconfigPaths(),
  ],
});

// react-router.config.ts
+import type { Config } from "@react-router/dev/config";
+export default {
+  ssr: true,
+} satisfies Config;
```

## 6. Add React Router plugin to `vite.config`

<docs-info>

If you used the codemod you can skip this step as it was automatically completed.

</docs-info>

**👉 Add `reactRouter` plugin to `vite.config`**

Change `vite.config.ts` to import and use the new `reactRouter` plugin from `@react-router/dev/vite`:

```diff

> -import { vitePlugin as remix } from "@remix-run/dev";

Reviewer: ```suggestion
-import { vitePlugin as remix } from "@remix-run/dev";
+import { reactRouter } from "@react-router/dev/vite";
```

);
```

## 8. Update types for `AppLoadContext`

<docs-info>

If you were using `remix-serve` you can skip this step. This is only applicable if you were using a custom server in Remix v2.

> Additionally, if you used the codemod you can skip this step as it was automatically completed.

Reviewer 2: Codemod doesn't handle updating AppLoadContext types.

Author: Whoops! Thanks for pointing that out, I'll remove it

> You can do this with the following code in your `vite.config.ts`:

Reviewer: ```suggestion
You can do this with the following code in `react-router.config.ts`:
```

Reviewer 3: We need to make sure that whatever file is used for `interface AppLoadContext` is included in your `tsconfig.json`'s `include`. I'm guessing most people will have `**/*.ts` which will cover this, but might be worth mentioning.

Author: Alright, added a comment calling this out, thanks @pcattori
